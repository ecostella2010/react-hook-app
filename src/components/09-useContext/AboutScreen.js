import React, { useContext } from 'react'
import { UseContext } from './UserContext';

export const AboutScreen = () => {

    const { user, setUser } = useContext ( UseContext );

    return (
        <div>
            <h1>AboutScreen</h1>
            <hr/>

            <pre>
                { JSON.stringify( user, null, 3 ) }
            </pre>

            <button
                className="btn btn-warning"
                onClick = { () => setUser ({}) }
            >
                Logout
            </button>


        </div>
    )
}
