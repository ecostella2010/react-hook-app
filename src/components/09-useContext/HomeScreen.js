import React, { useContext } from 'react'
import { UseContext } from './UserContext';

export const HomeScreen = () => {

    //const userContext = useContext ( UseContext );

    const { user } = useContext ( UseContext );

    console.log ( user );

    return (
        <div>
            <h1>HomeScreen</h1>
            <hr/>

            <pre>
                { JSON.stringify(user, null, 3) }
            </pre>
        </div>
    )
}
