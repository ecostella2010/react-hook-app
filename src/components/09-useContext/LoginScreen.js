import React, { useContext } from 'react'
import { UseContext } from './UserContext';

export const LoginScreen = () => {

    //1.- Obtener las referencias al userContext
    const { setUser } = useContext ( UseContext );
    //2.- Extraer el setUser de ese context

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr/>
            <button
                className="btn btn-primary"
                onClick ={ () => setUser( 
                    {
                        id: 1234,
                        name: 'Eduardo Costella',
                        email: 'ecostella@hotmail.com'
                    }
                 )}
            >
                Login
            </button>
        </div>
    )
}
