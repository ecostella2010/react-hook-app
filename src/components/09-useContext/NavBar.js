import React from 'react'
import { Link, NavLink } from 'react-router-dom';


export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
                <Link className="navbar-brand" to="/"> useContext </Link>
                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                    <div className="navbar-nav">
                        <NavLink exact="true" activeclassname="active" className="nav-link" aria-current="page" to="/">Home</NavLink>
                        <NavLink exact="true" activeclassname="active" className="nav-link" to="/about">About</NavLink>
                        <NavLink exact="true" activeclassname="active" className="nav-link" to="/login">Login</NavLink>
                    </div>
                </div>
        </nav>

        // <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
        // <Link className="navbar-brand" to="/">useContext</Link>
        // <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
        //     <div className="navbar-nav">
        //         <Link className="nav-link active" aria-current="page" to="/">Home</Link>
        //         <Link className="nav-link" to="/about">About</Link>
        //         <Link className="nav-link" to="/login">Login</Link>
        //     </div>
        // </div>
        // </nav>
        // <nav>
        //     <ul>
        //         <li>
        //             <Link to="/">Home</Link> 
        //         </li>
        //         <li>
        //             <Link to="/about">About</Link> 
        //         </li>
        //         <li>
        //             <Link to="/login">Login</Link> 
        //         </li>
        //     </ul>
        // </nav>
    )
}
