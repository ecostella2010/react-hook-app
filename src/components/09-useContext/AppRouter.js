import React from 'react';

import {
    BrowserRouter as Router,
    Routes as Switch, 
    Route,
    //Redirect,
    //Link
  } from 'react-router-dom';

import { NavBar } from './NavBar';
import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';


export const AppRouter = () => {
    return (
        <Router>
            <NavBar />
            <div className="container">
                <Switch>
                    <Route exact path="/" element={<HomeScreen />} />
                    <Route exact path="/about" element={<AboutScreen />} />
                    <Route exact path="/login" element={<LoginScreen />} />
                    <Route element={<HomeScreen />} />
                    {/* <Redirect to="/" /> */}
                </Switch>
            </div>
        </Router>
    )
}
