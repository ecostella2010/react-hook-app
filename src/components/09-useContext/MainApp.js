import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { UseContext } from './UserContext'


export const MainApp = () => {
    // const user = {
    //     id: 1234,
    //     name: 'Eduardo Costella',
    //     email: 'ecostella@hotmail.com'
    // }

    const [user, setUser] = useState({});

    return (
        <UseContext.Provider value = { {
            user, 
            setUser
        }}>
            <AppRouter />
        </UseContext.Provider>
    ) 
}
