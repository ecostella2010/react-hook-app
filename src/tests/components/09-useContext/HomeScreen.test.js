
import React from "react";
import { mount } from "enzyme"
import { HomeScreen } from "../../../components/09-useContext/HomeScreen";
import { UseContext } from "../../../components/09-useContext/UserContext";



describe('Pruebas en <HomeScreen/>', () => {
    const user = {
        id: 1234,
        name: 'Eduardo Costella',
        email: 'ecostella@hotmail.com'
    }
    
    //El shallow solo renderizar√° el primer componente, por lo que debemos usar mount
    const wrapper = mount (
        <UseContext.Provider value={{
            user
        }}>
            <HomeScreen />
        </UseContext.Provider> 
    
    );

    test('debe de mostrarse correctamente', () => {
        
        expect ( wrapper ).toMatchSnapshot ();
    });
    
})
