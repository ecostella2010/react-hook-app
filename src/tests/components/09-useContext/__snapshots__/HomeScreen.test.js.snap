// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen/> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "id": 1234,
   "name": "Eduardo Costella",
   "email": "ecostella@hotmail.com"
}
  </pre>
</div>
`;
